File: python/695.py
"""Project Euler Problem 695: Random Rectangles.

Choose three random points in the unit square and consider the three
rectangles with sides parallel to the unit square with each of the pairs of
points as diagonals. Find the expected median area of the three rectangles.

The solution involves integrating over satisfactory points P3 and computing
the expected value analytically.
"""

from __future__ import annotations

import math


def solve() -> float:
    """Solve Problem 695."""
    ans = (
        24 * math.log((3 + math.sqrt(5)) / 4)
        + 22 * math.sqrt(5)
        - 41
    ) / 144
    return ans


def main() -> float:
    """Main entry point."""
    result = solve()
    print(f"{result:.10f}")
    return result


if __name__ == "__main__":
    main()

File: c/695.c
/*
 * Project Euler 695 - Random Rectangles
 *
 * Expected median area of three rectangles formed by three random points
 * in the unit square.
 *
 * Closed-form: (24*ln((3+sqrt(5))/4) + 22*sqrt(5) - 41) / 144
 */
#include <stdio.h>
#include <math.h>

int main(void) {
    double s5 = sqrt(5.0);
    double ans = (24.0 * log((3.0 + s5) / 4.0) + 22.0 * s5 - 41.0) / 144.0;
    printf("%.10f\n", ans);
    return 0;
}
